Breakup deployments.adoc into subdir

adellape opened the pull request:
Preview build:

http://file.rdu.redhat.com/~adellape/110816/deployments_breakup/dev_guide/deployments/how_deployments_work.html

FYI @aheslin on this initial quick crack at chopping things up (haven't updated any xrefs or anything).

kargakis commented on dev_guide/deployments/advanced_deployment_strategies.adoc:
```diff
@@ -0,0 +1,295 @@
+. Trigger a re-deployment of the first shard to pick up the new labels:
++
+----
+$ oc deploy ab-example-a --latest
```
This changed recently to `oc rollout latest`. (All `deploy --latest` invocations actually changed to `rollout latest`)

adellape replied:
OK, looks like it's in 1.4/3.4: https://github.com/openshift/origin/pull/9750. The related docs card is tagged for 3.5: https://trello.com/c/0eoX2ANs. I'm gonna leave these as-is for now, and we can switch usage + add fuller `oc deploy latest` docs in a later PR?

kargakis replied:
The label in https://trello.com/c/uQP2AxXy/508-8-deploy-most-recently-available-image is wrong since it was added in 3.4. It would be nice to switch this back to `rollout latest` for 3.4 docs since we want people to start using `rollout` and deprecate `deploy`.

kargakis commented on dev_guide/deployments/basic_deployment_operations.adoc:
```diff
@@ -0,0 +1,469 @@
+To stream the logs of the latest revision for a given deployment configuration:
+
+----
+$ oc logs -f dc/<name> --follow
```
No need for calling follow twice.

kargakis commented on dev_guide/deployments/basic_deployment_operations.adoc:
```diff
@@ -0,0 +1,469 @@
+configuration `frontend` to 3.
+
+----
+$ oc scale dc frontend --replicas=3
```
Can you also add an example of setting up an autoscaler? `oc autoscale`. Follow-up to this PR is fine.

adellape replied:
Added a note box linking to the "Pod Autoscaling" topic for now.

kargakis commented:
+1 for breaking this up

adellape commented:
Updated per @kargakis comments and fixed all xrefs for new topics:

http://file.rdu.redhat.com/~adellape/110816/deployments_breakup/dev_guide/deployments/how_deployments_work.html

ahardin-rh commented:
@adellape This looks so much better. Thanks for reworking it. :trophy:  LGTM!

adellape commented:
[rev_history]
|Deployments
a|The previous _Deployments_ topic has been broken up into a subdirectory with multiple topics:
+
- xref:../../dev_guide/how_deployments_work.adoc#dev-guide-how-deployments-work[How Deployments Work]
- xref:../../dev_guide/basic_deployment_operations.adoc#dev-guide-basic-deployment-operations[Basic Deploment Operations]
- xref:../../dev_guide/deployment_strategies.adoc#dev-guide-deployment-strategies[Deployment Strategies]
- xref:../../dev_guide/deployments/advanced_deployment_strategies.adoc#dev-guide-advanced-deployment-strategies[Advanced Deployment Strategies]
%
